js_modules/dagit/packages/core/src/app/CustomConfirmationProvider.tsx

```diff
@@ -5,27 +5,33 @@ interface ConfirmationOptions {
   catchOnCancel?: boolean;
   title?: string;
   description?: React.ReactNode;
+  icon?: ConfirmationDialogProps['icon'];
+  intent?: ConfirmationDialogProps['intent'];
 }
 
 interface ConfirmationDialogProps extends ConfirmationOptions {
   open: boolean;
+  icon?: React.ComponentProps<typeof Dialog>['icon'];
+  intent?: React.ComponentProps<typeof Button>['intent'];
   onSubmit: () => void;
   onClose: () => void;
 }
 
 const ConfirmationDialog: React.FC<ConfirmationDialogProps> = ({
   open,
+  icon,
   title,
+  intent = 'danger',
   description,
   onSubmit,
   onClose,
 }) => {
   return (
-    <Dialog icon={title ? 'info' : undefined} onClose={onClose} title={title} isOpen={open}>
+    <Dialog icon={title ? icon ?? 'info' : icon} onClose={onClose} title={title} isOpen={open}>
```

Comment:
Hm, does `icon` need to care about the `title` value?

Reply:
I don't know but I figured I should preserve the existing behavior in case we're relying on it somewhjere